app/views/payment_gateways/_stripe_elements.html.erb

```diff
@@ -0,0 +1,146 @@
+                iconColor: '#e85746',
+                color: '#e85746',
+            }
+        },
+        classes: {
+            focus: 'is-focused',
+            empty: 'is-empty',
+        },
+    });
+    card.mount('#card-element');
+
+    var inputs = document.querySelectorAll('input.field');
+    Array.prototype.forEach.call(inputs, function(input) {
+        input.addEventListener('focus', function() {
+            input.classList.add('is-focused');
+        });
+        input.addEventListener('blur', function() {
+            input.classList.remove('is-focused');
+        });
+        input.addEventListener('keyup', function() {
+            if (input.value.length === 0) {
+                input.classList.add('is-empty');
+            } else {
+                input.classList.remove('is-empty');
+            }
+        });
+    });
+
+    function setOutcome(result) {
+        var errorElement = document.getElementById('card-errors');
+        if (result.token) {
+
+            var token = result.token;
+            var card = token.card;
+
+            // Insert the token ID into the form so it gets submitted to the server
+            var form = document.getElementById('stripe-form');
+            var hiddenInput = document.createElement('input');
+            hiddenInput.setAttribute('type', 'hidden');
+            hiddenInput.setAttribute('name', 'stripe[token]');
+            hiddenInput.setAttribute('value', token.id);
+            form.appendChild(hiddenInput);
+
+            var expiresOnMonth = document.createElement('input');
+            expiresOnMonth.setAttribute('type', 'hidden');
+            expiresOnMonth.setAttribute('name', 'stripe[expires_on_month]');
+            expiresOnMonth.setAttribute('value', card.exp_month);
+            form.appendChild(expiresOnMonth);
+
+            var expiresOnYear = document.createElement('input');
+            expiresOnYear.setAttribute('type', 'hidden');
+            expiresOnYear.setAttribute('name', 'stripe[expires_on_year]');
+            expiresOnYear.setAttribute('value', card.exp_year);
+            form.appendChild(expiresOnYear);
+
+            var partialNumber  = document.createElement('input');
+            partialNumber.setAttribute('type', 'hidden');
+            partialNumber.setAttribute('name', 'stripe[partial_number]');
+            partialNumber.setAttribute('value', card.last4);
+            form.appendChild(partialNumber);
+
+            // Submit the form
+            form.submit();
+        } else if (result.error) {
+            errorElement.textContent = result.error.message;
+            errorElement.classList.add('visible');
+        }
+    }
+
+    card.on('change', function(event) {
+        setOutcome(event);
+    });
+
+    document.getElementById('stripe-form').addEventListener('submit', function(e) {
+        e.preventDefault();
+        var extraDetails = {
+            address_line1: "<%= j current_account.billing_address.address1 %>",
```

Comment:
Minor thing and I wouldn't ask you to repush only for this, but imo it would be nice to access to `current_account.billing_address` only once 😄by saving it in a variable right before then accessing to that variable the whole time 😄

Reply:
OK I'll do